source/extensions/http/header_validators/envoy_default/header_validator.h

```diff
@@ -19,38 +22,94 @@ class HeaderValidator : public ::Envoy::Http::HeaderValidator {
+  /*
+   * Validate the :method pseudo header, honoring the restrict_http_methods configuration option.
+   */
+  virtual HeaderEntryValidationResult
+  validateMethodHeader(const ::Envoy::Http::HeaderString& value);
 
-  HeaderEntryValidationResult
-  validateResponseHeaderEntry(const ::Envoy::Http::HeaderString& key,
-                              const ::Envoy::Http::HeaderString& value) override;
+  /*
+   * Configuration for validateStatusHeader method.
+   */
+  enum class StatusPseudoHeaderValidationMode {
+    // Only accept whole number integer values.
+    WholeNumber,
 
-  RequestHeaderMapValidationResult
-  validateRequestHeaderMap(::Envoy::Http::RequestHeaderMap& header_map) override;
+    // Only accept values in the following range: 100 <= status <= 599.
+    ValueRange,
 
-  ResponseHeaderMapValidationResult
-  validateResponseHeaderMap(::Envoy::Http::ResponseHeaderMap& header_map) override;
+    // Only accept RFC registered status codes:
+    // https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml.
+    OfficialStatusCodes,
+  };
 
-protected:
-  StreamInfo::StreamInfo& stream_info_;
-};
+  /*
+   * Validate the :status response pseudo header.
+   */
+  virtual HeaderEntryValidationResult
+  validateStatusHeader(const StatusPseudoHeaderValidationMode& mode,
+                       const ::Envoy::Http::HeaderString& value);
 
-class HeaderValidatorFactory : public ::Envoy::Http::HeaderValidatorFactory {
-public:
-  HeaderValidatorFactory(
-      const envoy::extensions::http::header_validators::envoy_default::v3::HeaderValidatorConfig&
-          config);
+  /*
+   * Validate any request or response header name.
+   */
+  virtual HeaderEntryValidationResult
+  validateGenericHeaderName(const ::Envoy::Http::HeaderString& name);
+
+  /*
+   * Validate any request or response header value.
+   */
+  virtual HeaderEntryValidationResult
+  validateGenericHeaderValue(const ::Envoy::Http::HeaderString& value);
+
+  /*
+   * Validate the Content-Length request and response header as a whole number integer.
+   */
+  virtual HeaderEntryValidationResult
+  validateContentLengthHeader(const ::Envoy::Http::HeaderString& value);
 
-  ::Envoy::Http::HeaderValidatorPtr create(::Envoy::Http::Protocol protocol,
-                                           StreamInfo::StreamInfo& stream_info) override;
+  /*
+   * Validate the :scheme pseudo header.
+   */
+  virtual HeaderEntryValidationResult
+  validateSchemeHeader(const ::Envoy::Http::HeaderString& value);
 
-private:
+  /*
+   * Validate the Host header or :authority pseudo header. This method does not allow the
+   * userinfo component (user:pass@host).
+   */
+  virtual HeaderEntryValidationResult validateHostHeader(const ::Envoy::Http::HeaderString& value);
+
+  /*
+   * Validate the :path pseudo header. This method only validates that the :path header only
+   * contains valid characters and does not validate the syntax or form of the path URI.
```

Comment:
Where do we validate those?  I was checking for connect validation and didn't find it (yet)

maybe rename: validatePathHeaderCharacters

Reply:
This gets called in the H1 and H2 header validators in the `validateRequestHeaderEntry` methods. I'm renaming the method now.